docker.go

```diff
@@ -379,7 +379,18 @@ func (c *DockerContainer) ContainerIP(ctx context.Context) (string, error) {
 		return "", err
 	}
 
-	return inspect.NetworkSettings.IPAddress, nil
+	ip := inspect.NetworkSettings.IPAddress
+	if ip == "" {
+		// use IP from "Networks" if only single network defined
+		networks := inspect.NetworkSettings.Networks
+		if len(networks) == 1 {
```

Comment:
If len is 1, then I think we can avoid the for loop and directly get the index 0 of the array.

Reply:
At the same time, we could create the convention that it will always return the IP from the first network

Reply:
The problem here is that `networks` is a map, not a slice. Then, I think you can't just take the index 0. Also, taking the always the first network is problematic, because it can easily become non-deterministic. Then, I'd just limit it to returning a value if `len(networks) == 1`.

Btw. I think in the future we could create a function to return all IP addresses, instead of just one.

Reply:
Ah ah thanks for the explanation, then it makes sense to do it as you suggest.